server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -506,138 +290,348 @@ public boolean deleteLookup(final String tier, final String lookup, AuditInfo au
-      );
-      Futures.addCallback(backgroundManagerFuture, new FutureCallback<Object>()
-      {
-        @Override
-        public void onSuccess(@Nullable Object result)
-        {
-          backgroundManagerExitedLatch.countDown();
-          LOG.debug("Exited background lookup manager");
+        );
+
+        configManager.set(
+            LOOKUP_CONFIG_KEY,
+            converted,
+            new AuditInfo("autoConversion", "autoConversion", "127.0.0.1")
+        );
+      }
+    }
+  }
+
+  private Map<String, LookupExtractorFactoryMapContainer> convertTierLookups(
+      Map<String, Map<String, Object>> oldTierLookups
+  )
+  {
+    Map<String, LookupExtractorFactoryMapContainer> convertedTierLookups = new HashMap<>();
+    oldTierLookups.forEach(
+        (lookup, lookupExtractorFactory) -> {
+          convertedTierLookups.put(lookup, new LookupExtractorFactoryMapContainer(null, lookupExtractorFactory));
         }
+    );
+    return convertedTierLookups;
+  }
 
-        @Override
-        public void onFailure(Throwable t)
-        {
-          backgroundManagerExitedLatch.countDown();
-          if (backgroundManagerFuture.isCancelled()) {
-            LOG.info("Background lookup manager exited");
-            LOG.trace(t, "Background lookup manager exited with throwable");
-          } else {
-            LOG.makeAlert(t, "Background lookup manager exited with error!").emit();
-          }
+  @VisibleForTesting
+  void lookupManagementLoop()
+  {
+    // Sanity check for if we are shutting down
+    if (Thread.currentThread().isInterrupted() || !lifecycleLock.awaitStarted(15, TimeUnit.SECONDS)) {
+      LOG.info("Not updating lookups because process was interrupted or not finished starting yet.");
+      return;
+    }
+
+    final Map<String, Map<String, LookupExtractorFactoryMapContainer>> allLookupTiers = lookupMapConfigRef.get();
+
+    if (allLookupTiers == null) {
+      LOG.info("Not updating lookups because no data exists");
+      return;
+    }
+
+    LOG.debug("Starting lookup sync for on all nodes.");
+
+    final Map<HostAndPort, LookupsState<LookupExtractorFactoryMapContainer>> currState = new ConcurrentHashMap<>();
+
+    try {
+      List<ListenableFuture<Map.Entry>> futures = new ArrayList<>();
+      for (Map.Entry<String, Map<String, LookupExtractorFactoryMapContainer>> tierEntry : allLookupTiers.entrySet()) {
+
+        LOG.debug("Starting lookup mgmt for tier [%s].", tierEntry.getKey());
+
+        final Map<String, LookupExtractorFactoryMapContainer> tierLookups = tierEntry.getValue();
+        for (final HostAndPort node : listenerDiscoverer.getNodes(LookupModule.getTierListenerPath(tierEntry.getKey()))) {
+
+          LOG.debug(
+              "Starting lookup mgmt for tier [%s] and host [%s:%s].",
+              tierEntry.getKey(),
+              node.getHostText(),
+              node.getPort()
+          );
+
+          futures.add(
+              executorService.submit(
```

Comment:
For the coordinator management epoch, we could use `executorService` as the indicator. Meaning the runnable aka management loop can take it as a parameter, then when the executor service shuts down during the stop, this will fail with a rejected execution exception and the management can end. Otherwise we can have a runnable submitting to a different epoch's executor service if I'm reading this correctly.